### DISCUSSION TITLE
Use Case Ideas

### DISCUSSION THREAD

**Discussion by RyanStrange** [SENG-350-2024-fall/Team12#1] [Milestone 1]
- register
- login?
- check wait time
- virtual triage
- find nearest ER

**Comment by RyanStrange** [comment]
https://www.geeksforgeeks.org/use-case-diagram/

use case tutorial

**Comment by MateyaMSB** [comment]
Use Case: Register with Mister Ed
Description: Enter your name,health care number,and location,and contact information in registration page
Actors: patients
Assumptions: User have a means of accessing Mister Ed (Wifi, computer)
Steps:

1. open mister Ed
2. navigate to registration page
3. enter name
4. enter health card number
5. enter location
6. enter preferred contact information
7. save
Variations: maybe no location, maybe no health care number

**Comment by MateyaMSB** [comment]
create account -> login-> check availability of nearby ED's ->  perform triage -> enter queue if you need to go to ED  or check alternative solution if fail triage-> check how long your wait time is ->have alternative ways to be contacted ( through Mister Ed and personal phone)

**Comment by MateyaMSB** [comment]
Use cases:
create account/login
check availability 
perform triage
check alternatives
enter queue/ check your wait time
alert of availibility

**Reply by ellucid** [reply]
Perform Triage - Josephine

**Reply by nahirv** [reply]
working on check availability

**Reply by RyanStrange** [reply]
create account/login - Ryan

**Reply by MateyaMSB** [reply]
Mateya - check alternatives

**Comment by ellucid** [comment]
Use Case: Virtual Triage
Description: Patients are prompted to answer questions regarding symptoms to determine next steps
Actors: Patients, Nurse/Specialist/AI
Assumptions: Nurse/Specialist is available, or else a chatbot is used
Steps:
1. Start virtual triage
2. Answer questions from nurse/specialist or chatbot
3. Upon virtual triage completion, take next recommended steps
3.a go to a GP
3.b take over-the-counter medication
3.c contact a nurse
3.d attend in person

**Comment by nahirv** [comment]
Use case: Check availability
Description: View current load of Emergency Departments in the area
Actors: Patients
Assumptions: Mister Ed system has access to real time data from EDs, patient can access Mister Ed system
Steps: 
1. Patient navigates to current loads page
2. Patient allows system to access their location
3. Patient views availability of the nearest EDs
4. Patient proceeds to virtual triage if not done already

Variations: 2.Patient may allow system to use their location or manually enter their location
4.Patient may undergo virtual triage or simply exit page
Non-Functional: Performance: The system must display availability information in less than 30 seconds. 
Reliability:  Users must be able to view current ED loads at all times.
Frequency: The system must update ED information every 5 minutes.
Issues: What will the system do if data from an ED is unavailable?

**Comment by MateyaMSB** [comment]
![Use-case-diagram](https://github.com/user-attachments/assets/be2fa902-0fd8-4e05-a8c9-9700d0c727d9)

**Comment by ellucid** [comment]
Examples of existing VT system
https://support.evisit.com/hc/en-us/categories/360006231693-eVisit-Support

**Comment by MateyaMSB** [comment]
@startuml
left to right direction
actor "Patient" as p
rectangle "MisterED" {
  usecase "Complete Virtual Triage" as UC3
  usecase "Enter Queue" as UC2
  usecase "See Alternative Health Solutions" as UC6
  usecase "Check ER Capacities" as UC4
  usecase "Create Account/Log In" as UC1
  usecase "Check Queue Status" as UC5
}
p --> UC1
UC3 .> UC2 : <<includes>>
p --> UC3
p --> UC4
UC3 .> UC6 : <<includes>>
UC2 .> UC5 : <<includes>>
p --> UC2
p --> UC6
p--> UC5

@enduml

**Comment by MateyaMSB** [comment]
hi
